### ISSUE TITLE
Plumb the proxy dialer to the webhook admission plugin

### ISSUE THREAD

**Issue by caesarxuchao** [kubernetes/kubernetes#49987]
Currently webhoook admission plugin doesn't have a dialer: https://github.com/kubernetes/kubernetes/blob/master/plugin/pkg/admission/webhook/admission.go#L260

Probably we need to plumb the dialer to the client like aggregator does: https://github.com/kubernetes/kubernetes/blob/master/staging/src/k8s.io/kube-aggregator/pkg/apiserver/handler_proxy.go#L213. Otherwise the plugin won't be able to work in gke.

cc @mandarjog @cheftako

**Comment by mandarjog** [comment]
On GKE 1.7.2
```
W0801 14:00:18.957110       4 admission.go:174] Failed calling webhook pod-image.k8s.io: failed calling admission webhook "pod-image.k8s.io": Post https://10.40.0.5:8000/: net/http: request canceled while waiting for connection (Client.Timeout exceeded while awaiting headers)
E0801 14:00:18.957140       4 admission.go:175] failed calling admission webhook "pod-image.k8s.io": Post https://10.40.0.5:8000/: net/http: request canceled while waiting for connection (Client.Timeout exceeded while awaiting headers)
```
API server attempts to open direct connections to endpoints.

**Comment by caesarxuchao** [comment]
I'll look into the issue this week.

**Comment by mandarjog** [comment]
There is a workaround.
1. Deploy webhook as a `LoadBalancer` type service and get an external IP.
2. Deploy webhookExternal like so
```
apiVersion: v1
kind: Service
metadata:
  name: webhookExternal
spec:
  externalIPs:
  - 35.193.46.50
  ports:
    - port: 443
      protocol: TCP
```

Create and endpoint object named `webhookExternal`
```
apiVersion: v1
kind: Endpoints
metadata:
  name: webhookExternal
  namespace: default
subsets:
- addresses:
  - ip: 35.193.46.50
  ports:
  - port: 443
    protocol: TCP
```

Then use `webhookExternal` in the validator definition.

**Comment by caesarxuchao** [comment]
@mandarjog I don't have time testing the fix (#50476) before code freeze. I'll resume working on this after Sept.1. The fix will be included in 1.8.

**Comment by dims** [comment]
@caesarxuchao Looks like at least one PR merged for this. Can we close this? or push it out of 1.8 milestone please?

**Comment by mbohlool** [comment]
@cheftako (as @caesarxuchao is OOO): this is a daily and friendly ping from release team. Is this still relevant to 1.8? If yes, who is actively working on it?

**Comment by lavalamp** [comment]
I am pretty sure @caesarxuchao fixed this in #50476.

**Comment by cheftako** [comment]
As Chao mentions https://github.com/kubernetes/kubernetes/pull/50476 should fix the issue. He manually tested that this works. The only thing that I know of that we are missing for this is an automated e2e test for GKE to make sure we don't get a regression.
